handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/GetAccount.scala

```diff
@@ -0,0 +1,112 @@
+import scala.util.Try
+
+object GetAccountLive:
+  val layer: URLayer[ZuoraGet, GetAccount] = ZLayer.fromFunction(GetAccountLive(_))
+
+private class GetAccountLive(zuoraGet: ZuoraGet) extends GetAccount :
+  override def get(accountNumber: String): IO[String, GetAccountResponse] =
+    zuoraGet.get[GetAccountResponse](uri"accounts/$accountNumber/summary")
+
+  override def getPaymentMethod(paymentMethodId: String): IO[String, PaymentMethodResponse] =
+    zuoraGet.get[PaymentMethodResponse](uri"object/payment-method/$paymentMethodId")
+
+trait GetAccount:
+  def get(subscriptionNumber: String): ZIO[GetAccount, String, GetAccountResponse]
+
+  def getPaymentMethod(paymentMethodId: String): ZIO[GetAccount, String, PaymentMethodResponse]
+
+object GetAccount {
+  case class PaymentMethodResponse(NumConsecutiveFailures: Int)
+
+  given JsonDecoder[PaymentMethodResponse] = DeriveJsonDecoder.gen[PaymentMethodResponse]
+
+  case class GetAccountResponse(
+    basicInfo: BasicInfo,
+    subscriptions: List[AccountSubscription],
+  )
+
+  case class AccountSubscription(
+    id: String
+  )
+
+  case class BasicInfo(
+    defaultPaymentMethod: WireDefaultPaymentMethod,
+    balance: BigDecimal,
+    currency: String
+  )
+
+  /* The zuora API has slightly different responses for ratePlans in the catalogue and when querying a subscription, e.g.: productRatePlanId in the subscription response and id in the catalogue response */
+
+  case class ZuoraSubscription(
+    ratePlans: List[ZuoraRatePlan]
+  )
+
+  case class ZuoraRatePlan(
+    productRatePlanId: String // TBC what field should we check in the rateplan to know it's a recurring contribution?
+    /*
+    {
+                    "productId": "2c92c0f84b786da2014b91d3629b4298",//DEV
+                    "productName": "Digital Pack",
+                    "productSku": "SKU-00000022",
+                    "productRatePlanId": "2c92c0f94bbffaaa014bc6a4212e205b",//DEV
+                    "ratePlanName": "Digital Pack Annual"
+                }*/
+  )
+
+  given JsonDecoder[GetAccountResponse] = DeriveJsonDecoder.gen
+  given JsonDecoder[AccountSubscription] = DeriveJsonDecoder.gen
+  given JsonDecoder[ZuoraSubscription] = DeriveJsonDecoder.gen
+  given JsonDecoder[ZuoraRatePlan] = DeriveJsonDecoder.gen
+  given JsonDecoder[BasicInfo] = DeriveJsonDecoder.gen
+
+  def get(accountNumber: String): ZIO[GetAccount, String, GetAccountResponse] =
+    ZIO.serviceWithZIO[GetAccount](_.get(accountNumber))
+
+  def getPaymentMethod(paymentMethodId: String): ZIO[GetAccount, String, PaymentMethodResponse] = ZIO.serviceWithZIO[GetAccount](_.getPaymentMethod(paymentMethodId))
+}
+
+case class DefaultPaymentMethod(
+  id: String,
+  creditCardExpirationDate: LocalDate,
+)
+
+case class WireDefaultPaymentMethod(
+  id: String,
+  creditCardExpirationMonth: Option[Int],
+  creditCardExpirationYear: Option[Int],
+)
+
+given JsonDecoder[WireDefaultPaymentMethod] = DeriveJsonDecoder.gen[WireDefaultPaymentMethod]
```

Comment:
we can keep all the wrangling in the decoder rather than the business logic something like this:
```suggestion
object DefaultPaymentMethod {
  private case class WireDefaultPaymentMethod(
    id: String,
    creditCardExpirationMonth: Option[Int],
    creditCardExpirationYear: Option[Int],
  )

  given JsonDecoder[DefaultPaymentMethod] = DeriveJsonDecoder.gen[WireDefaultPaymentMethod].map {
    case WireDefaultPaymentMethod(id, month, year) =>
      DefaultPaymentMethod(id, LocalDate.of(creditCardExpirationYear, creditCardExpirationMonth, 1))
  }
}
```

Reply:
Where would we perform the Option handling for this?

Reply:
updated this during pairing session